File: src/main/java/com/example/twitterapi/service/TweetService.java
package com.example.twitterapi.service;

import com.example.twitterapi.dto.request.TweetCreateRequest;
import com.example.twitterapi.dto.request.TweetUpdateRequest;
import com.example.twitterapi.dto.response.TweetResponse;
import java.util.List;

public interface TweetService {

    TweetResponse createTweet(String username, TweetCreateRequest request);

    List<TweetResponse> getAllTweets();

    List<TweetResponse> getTweetsByUserId(Long userId);

    TweetResponse getTweetById(Long id);

    TweetResponse updateTweet(Long id, String username, TweetUpdateRequest request);

    void deleteTweet(Long id, String username);
}

File: frontend/src/services/tweetService.js
import { request } from './apiClient';

export function createTweet(body, auth) {
  return request('/tweet', {
    method: 'POST',
    body,
    auth
  });
}

export function getAllTweets(auth) {
  return request('/tweet', {
    auth
  });
}

export function getTweetsByUserId(userId, auth) {
  return request(`/tweet/findByUserId?userId=${userId}`, {
    auth
  });
}

export function getTweetById(tweetId, auth) {
  return request(`/tweet/findById?id=${tweetId}`, {
    auth
  });
}

export function updateTweet(tweetId, body, auth) {
  return request(`/tweet/${tweetId}`, {
    method: 'PUT',
    body,
    auth
  });
}

export function deleteTweet(tweetId, auth) {
  return request(`/tweet/${tweetId}`, {
    method: 'DELETE',
    auth
  });
}
